app/home/auth-factory.1.coffee

```diff
@@ -0,0 +1,90 @@
+'use strict'
+
+###*
+ # @ngdoc service
+ # @name home.factory:Auth
+
+ # @description Authentication service for manage users (NEW)
+
+###
+angular
+  .module 'home'
+  .factory 'AuthFactory1',['$firebaseAuth','$state', '$mdToast', '$http', '$q', ($firebaseAuth,$state, $mdToast, $http, $q) ->
+    new class Auth
+      constructor: () ->
+        @url = "https://examsdash.firebaseio.com/users.json"
+        @ref = new Firebase ("https://examsdash.firebaseio.com/")
+        @auth = $firebaseAuth(@ref)
+        @error = null
+        @loggedIn = @ref.getAuth()?
+        @message = null
+        if @ref.getAuth()? then @getUserData(@ref.getAuth().uid) else @user={}
+        @showToast = showToast
+      showToast = (message) ->
+        $mdToast.show(
+          $mdToast.simple()
+            .textContent(message)
+            .position("top right")
+            .hideDelay(3000)
+        )
+      getUserData: (uid) =>
+        deferred = $q.defer()
+        request = $http.get @url
+        request.then (result) =>
+          @user=result.data[uid]
+      getUserinoData = (uid, url) ->
+        deferred = $q.defer()
+        request = $http.get url
+        request.then (result) ->
```

Comment:
I always prefer named methods in order to debug is just a sugestion. I would create onSuccessGetUserInfoData and so on...